report/id.go

```diff
@@ -29,25 +30,37 @@ const (
 
 // MakeEndpointNodeID produces an endpoint node ID from its composite parts.
 func MakeEndpointNodeID(hostID, namespaceID, address, port string) string {
-	return makeAddressID(hostID, namespaceID, address) + ScopeDelim + port
+	addressIP := net.ParseIP(address)
+	return makeAddressID(hostID, namespaceID, address, addressIP) + ScopeDelim + port
+}
+
+// MakeEndpointNodeIDB produces an endpoint node ID from its composite parts in binary, not strings.
+func MakeEndpointNodeIDB(hostID string, namespaceID uint64, addressIP net.IP, port uint16) string {
+	namespace := strconv.FormatUint(namespaceID, 10)
+	return makeAddressID(hostID, namespace, addressIP.String(), addressIP) + ScopeDelim + strconv.Itoa(int(port))
 }
 
 // MakeAddressNodeID produces an address node ID from its composite parts.
 func MakeAddressNodeID(hostID, address string) string {
-	return makeAddressID(hostID, "", address)
+	addressIP := net.ParseIP(address)
+	return makeAddressID(hostID, "", address, addressIP)
 }
 
-func makeAddressID(hostID, namespaceID, address string) string {
+// MakeAddressNodeIDB produces an address node ID from its composite parts, in binary not string.
+func MakeAddressNodeIDB(hostID string, addressIP net.IP) string {
+	return makeAddressID(hostID, "", addressIP.String(), addressIP)
+}
+
+func makeAddressID(hostID, namespaceID, address string, addressIP net.IP) string {
```

Comment:
This is one of those places where we were converting back and forward between string and binary form, so passing both is expected to be an improvement.

Reply:
I was a bit confused looking at this because I expected that they probably contain different _data_ as both are passed as args, instead of just being of different _type_. Maybe a one line comment about avoiding excessive conversion would be helpful here as in the usual circumstances code brevity/readability would probably be preferred to reducing the conversions.

Reply:
I'm going to refactor this in a subsequent update, so leaving it as-is for now.